### PR TITLE
Allow user to specify custom file extensions

### PR DESCRIPTION
**Pull request by Morriar**
### Motivation

Sometimes your code is located in files that are neither `.rb` or `.rbi`. For example with `.rake` or `.ru`.

This pull-request makes possible for the user to specify which extensions should be considered by sorbet-static thanks to the `--allowed-extension`.

Behavior:

* The default behaviour is unchanged: not passing any `--allowed-extension` results in Sorbet reading only `.rb` and `.rbi` files.
* Passing at least one `--allowed-extension` voids the default.
* The user can pass more than one `--allowed-extension` as to allow multiple custom extensions

Example:

```bash
$ sorbet \
--allowed-extension .rb    \
--allowed-extension .rbi   \
--allowed-extension .ru    \
--allowed-extension .rake  \
my_directory
```

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by Morriar** [comment]
Rerolled to please the linter.

**Comment by Morriar** [comment]
@jez the order the errors are displayed doesn't seem to be stable. How do you normally deal with this?

File: main/options/options.cc [hunk 1]
```diff
@@ -623,6 +624,14 @@ void readOptions(Options &opts,
             Exception::raise("simulated crash");
         }
 
+        if (raw.count("allowed-extension") > 0) {
+            auto exts = raw["allowed-extension"].as<vector<string>>();
```
Comment (paracycle) [inline main/options/options.cc]: I would have expected `--allowed-extension` to have placed the given extension _on top of_ the `.rb` and `.rbi` extensions. I understand this is more flexible but is there any case where someone would not want to have `.rb` and `.rbi` extensions?

Comment (Morriar) [reply main/options/options.cc]: Yes: me when testing and debugging :)

I sometimes even need to move / rename a RBI file to check things.

Comment (Morriar) [reply main/options/options.cc]: This way felt more elegant than: `--no-default-extensions --allowed-extension .ru`.

**Comment by Morriar** [comment]
Rerolled: I added some magic to the test so the files are always outputted in the same order.

**Comment by DarkDimius** [comment]
Thank you!

**Pull request merged**